client/src/pages/ApplicationsPage/ApplicationsPage.tsx

```diff
@@ -5,15 +5,32 @@ import { useAppSelector } from '../../app/hooks';
 import { IApplication } from '../../../types/applications';
 import ApplicationDashboard from '../../components/ApplicationDashBoard/ApplicationDashBoard';
 
-const ApplicationsPage = (): JSX.Element => {
+interface Params {
+  userId: string;
+  status?: string;
+  date?: string;
+}
+
+const ApplicationsPage = () => {
   const navigate = useNavigate();
   const [applications, setApplications] = useState<IApplication[]>([]);
+  const [showRejected, setShowRejected] = useState(true);
+  const [dateFilter, setDateFilter] = useState(false);
   const user = useAppSelector((state) => state.user.userData);
 
   useEffect(() => {
     async function fetchApplications() {
       try {
-        const response = await axios.get(`/api/applications?user_id=${user?._id}`);
+        const params: Params = { userId: user?._id ?? '' };
+        if (!showRejected) params.status = 'Rejected';
+
+        //TODO adjust time delay for production - Let user select dif times from dropdown?
+        if (dateFilter) {
+          const thirtySecondsAgo = new Date(Date.now() - 30 * 3000);
+          params.date = thirtySecondsAgo.toISOString();
+        }
+
+        const response = await axios.get(`/api/applications`, { params });
 
         setApplications(response.data);
```

Comment:
We fetch all the user's application data on page render - so the all the data is already client side. I'm wondering if it would be better to apply the filters client side vs. refetching and running another query on the DB?

Reply:
It will be. This is just temporary until migrated to Redux

Reply:
gotti